contracts/NodeDelegatorETH.sol

```diff
@@ -0,0 +1,411 @@
+
+        // Not adding any consensus rewards that have been sent to the EigenPod.
+        // This can include forced validator withdrawals so that needs to be accounted for
+        // in a future implementation.
+    }
+
+    /// @notice Stakes WETH or ETH in the NDC to multiple validators connected to an EigenPod.
+    /// @param validators A list of validator data needed to stake.
+    /// The ValidatorStakeData struct contains the pubkey, signature and depositDataRoot.
+    /// @dev Only accounts with the Operator role can call this function.
+    function stakeEth(ValidatorStakeData[] calldata validators) external whenNotPaused nonReentrant onlyLRTOperator {
+        // Yield from the validators will come as native ETH.
+        uint256 ethBalance = address(this).balance;
+        uint256 requiredETH = validators.length * 32 ether;
+        if (ethBalance < requiredETH) {
+            // If not enough native ETH, convert WETH to native ETH
+            uint256 wethBalance = IWETH(WETH).balanceOf(address(this));
+            if (wethBalance + ethBalance < requiredETH) {
+                revert InsufficientWETH(wethBalance + ethBalance);
+            }
+            // Convert WETH asset to native ETH
+            IWETH(WETH).withdraw(requiredETH - ethBalance);
+        }
+
+        // For each validator
+        for (uint256 i = 0; i < validators.length;) {
+            bytes32 pubkeyHash = keccak256(validators[i].pubkey);
+
+            if (validatorsStaked[pubkeyHash]) {
+                revert ValidatorAlreadyStaked(validators[i].pubkey);
+            }
+
+            _stakeEth(validators[i].pubkey, validators[i].signature, validators[i].depositDataRoot);
+            validatorsStaked[pubkeyHash] = true;
+
+            unchecked {
+                ++i;
+            }
+        }
+    }
+
+    /// @dev Stake WETH and ETH in NDC in EigenLayer. It calls the `stake` function on the EigenPodManager
+    /// which calls `stake` on the EigenPod contract which calls `stake` on the Beacon DepositContract.
+    /// @dev The public functions that call this internal function are responsible for access control.
+    function _stakeEth(bytes calldata pubkey, bytes calldata signature, bytes32 depositDataRoot) internal {
+        // Call the stake function in the EigenPodManager
+        IEigenPodManager eigenPodManager = IEigenPodManager(lrtConfig.getContract(LRTConstants.EIGEN_POD_MANAGER));
+        eigenPodManager.stake{ value: 32 ether }(pubkey, signature, depositDataRoot);
+
+        // Increment the staked but not verified ETH
+        stakedButNotVerifiedEth += 32 ether;
+
+        emit ETHStaked(pubkey, 32 ether);
+    }
+
+    /// @notice Delegates all staked assets from this contract to an EigenLayer Operator.
+    /// This includes both both LSTs and native ETH.
+    /// @param operator the address of the EigenLayer Operator to delegate to.
+    function delegateTo(address operator) external onlyLRTManager {
```

Comment:
Since we are sunsetting, and this contract should not hold anything but WETH, should this function just revert?